src/main/java/com/assignment1/item/ItemEntity.java

```diff
@@ -0,0 +1,200 @@
+  }
+
+  /**
+   * Setter.
+   */
+  public void setTax(final BigDecimal tax) {
+    this.tax = tax;
+  }
+
+  /**
+   * Overloading toString().
+   */
+  @Override
+  public String toString() {
+    return name + price + quantity + type;
+  }
+
+  /**
+   * method that calculates the tax.
+   */
+  public void calculateTax() {
+    switch (type) {
+      case RAW:
+        //raw tax: 12.5% of the item cost
+        tax = price.multiply(BigDecimal.valueOf(0.125));
+        break;
+      case MANUFACTURED:
+        //manufactured tax: 12.5% of the item cost + 2% of (item cost + 12.5% of the item cost)
+        tax = price.multiply(BigDecimal.valueOf(0.125));
+        tax = tax.add(price.add(tax).multiply(BigDecimal.valueOf(0.02)));
+        break;
+      case IMPORTED:
+        final BigDecimal surcharge100 = BigDecimal.valueOf(100);
+        final BigDecimal surcharge200 = BigDecimal.valueOf(200);
+        //imported: 10% import duty on item cost + a surcharge
+        // (surcharge is: Rs. 5 if the final cost after applying tax &
+        // import duty is up to Rs. 100, Rs. 10 if the cost exceeds 100 and up to 200
+        // and 5% of the final cost if it exceeds 200).
+        final BigDecimal importDuty = price.multiply(BigDecimal.valueOf(0.1));
+        //Further additional surcharge to be added according to final cost
+        final BigDecimal finalCost = price.add(importDuty);
+        if (finalCost.compareTo(surcharge100) <= 0) {
+          tax = finalCost.add(BigDecimal.valueOf(5));
+        } else if (finalCost.compareTo(surcharge100) > 0
+                && finalCost.compareTo(surcharge200) <= 0) {
+          tax = finalCost.add(BigDecimal.valueOf(10));
+        } else if (finalCost.compareTo(surcharge200) > 0) {
+          tax = finalCost.add(finalCost.multiply(BigDecimal.valueOf(0.05)));
+        }
+        tax = tax.subtract(price); //subtract price to have only tax, not final price
+        break;
+      default:
+        tax = BigDecimal.valueOf(0);
+        break;
+    }
+  }
+
+  /**
+   * method to read the item details and return string array for parsing
+   */
+  public ItemDTO read() throws IOException {
+    final ItemDTO itemDTO = new ItemDTO();
+    final BufferedReader bufferedReader =
+            new BufferedReader(new InputStreamReader(System.in));
+    System.out.print("Enter Name: ");
+    itemDTO.setName(bufferedReader.readLine());
+    System.out.print("Enter Price: ");
+    itemDTO.setPrice(bufferedReader.readLine());
+    System.out.print("Enter Quantity: ");
+    itemDTO.setQuantity(bufferedReader.readLine());
+    System.out.print("Enter Type: ");
+    itemDTO.setType(bufferedReader.readLine());
+    return itemDTO;
+  }
+
+  /**
+   * method to write the item details.
+   */
+  public void write() {
+    System.out.println(name
+            + ", " + price
+            + ", " + tax
+            + ", " + quantity
+            + ", " + BigDecimal.valueOf(quantity).multiply(price.add(tax)));
+  }
```

Comment:
move this to a separate class which implements an interface

Reply:
Made separate interfaces for these behaviors.